docs/examples/vault-support/setup.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -euo pipefail
+
+RABBITMQ_NAMESPACE=${RABBITMQ_NAMESPACE:-'default'}
+
+vault_exec () {
+    kubectl exec vault-0 -c vault -- /bin/sh -c "$*"
+}
+
+echo "Installing Vault server and Vault agent injector..."
+helm repo add hashicorp https://helm.releases.hashicorp.com
+helm repo update
+# For OpenShift deployments, also set the following:
+# --set "global.openshift=true"
+helm install vault hashicorp/vault \
+    --set='server.dev.enabled=true' \
+    --set='server.logLevel=debug' \
+    --set='injector.logLevel=debug' \
+    --wait
+sleep 5
+kubectl wait --for=condition=Ready pod/vault-0
+
+echo "Configuring K8s authentication..."
+# Required so that Vault init container and sidecar of RabbitmqCluster can authenticate with Vault.
+vault_exec "vault auth enable kubernetes"
+
+# In Kubernetes 1.21+ clusters, issuer may need to be configured as described in https://www.vaultproject.io/docs/auth/kubernetes#discovering-the-service-account-issuer
+# Otherwise, vault-agent-init container will output "error authenticating".
+issuer=$(kubectl get --raw=http://127.0.0.1:8001/.well-known/openid-configuration | jq -r .issuer)
+vault_exec "vault write auth/kubernetes/config issuer=\"$issuer\" token_reviewer_jwt=\"\$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" kubernetes_host=https://\${KUBERNETES_PORT_443_TCP_ADDR}:443 kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+#vault_exec "vault write auth/kubernetes/config token_reviewer_jwt=\"\$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" kubernetes_host=https://\${KUBERNETES_PORT_443_TCP_ADDR}:443 kubernetes_ca_cert=@/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+
+# Each RabbitMQ cluster may have its own secret path.
+echo "Creating credentials for rabbitmq default user in cluster-vault-a..."
+vault_exec "vault kv put secret/rabbitmq/cluster-vault-a/creds username='rabbitmq' password='pwda'"
+#echo "Creating credentials for rabbitmq default user in cluster-vault-b..."
+#vault_exec "vault kv put secret/rabbitmq/cluster-vault-b/creds username='rabbitmq' password='pwdb'"
+
+# Create a policy that allows reading of the default user credentials for the RabbitMQ cluster cluster-vault-a
+# The path must be referenced from the RabbitmqCluster cluster-vault-a CRD spec.secretBackend.vault.defaultUserPath
+echo "Creating Vault policy named cluster-vault-a-policy for reading of cluster-vault-a credentials..."
+vault_exec "vault policy write cluster-vault-a-policy - <<EOF
```

Comment:
I wonder whether it is useful to add a section to goes into suggesting ways to organize the authorization strategies.
In this example, we can suggesting a strategy whereby each cluster gets its own policy and role. It offers a great deal of security but i am not sure if it is scalable. 

Another approach is to let a group of clusters access by using wildcards in paths. For instance, have a policy which grants access to the path `secret/data/rabbitmq/finance-group/*`. A cluster whose name is "orders" would retrieve its credentials from this path `secret/data/rabbitmq/finance-group/creds`. And all the clusters in that group must use the same role, e.g. `finance-group-role`. Any time we create a cluster, we have to include the cluster`s service account in the role's `bound_service_account_names` attribute.